bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -786,14 +830,15 @@ public static Long win32_getHandle (Image image, int zoom) {
 		}
 		if (image.dataAtBaseZoom != null) {
 			ImageData resizedData = image.getImageData(zoom);
-			image.init(resizedData, zoom);
+			ImageData newData = image.copyImageDataForDisabledOrGrayImages(image.getDevice(), resizedData, image.styleFlag);
+			image.init(newData, zoom);
 			image.init();
 		}
 	}
 	return image.zoomLevelToHandle.get(zoom);
 }
 
-long initNative(String filename, int zoom) {
+long initNative(String filename, int zoom, int imageStyle) {
```

Comment:
Can you please explain a concrete scenario in which passing the `imageStyle` is necessary compared to just using the existing style of the image?

Reply:
As I mentioned in the comment above, `win32_getHandle `has a different object for an image. We need `imageStyle `of that particular image object which is different from `this`

Reply:
I don't get that one. Yes ,`win32_getHandle` calls this method for another image, but the metohd is then executed on exactly that instance, so you can simply use the values from that instance.

Reply:
@ShahzaibIbrahim as a rule of thumb: if the method is `static` then pass the state as parameters and if it's not then use the internal state (fields) :-)